packages/@aws-cdk/aws-autoscaling/lib/auto-scaling-group.ts

```diff
@@ -186,4 +249,213 @@ export class AutoScalingGroup extends cdk.Construct implements ec2.IClassicLoadB
+     * PT#H#M#S, where each # is the number of hours, minutes, and seconds,
+     * respectively). The maximum PauseTime is one hour (PT1H).
+     *
+     * @default 300 if the waitOnResourceSignals property is true, otherwise 0
+     */
+    pauseTimeSec?: number;
+
+    /**
+     * Specifies whether the Auto Scaling group waits on signals from new instances during an update.
+     *
+     * AWS CloudFormation must receive a signal from each new instance within
+     * the specified PauseTime before continuing the update.
+     *
+     * To have instances wait for an Elastic Load Balancing health check before
+     * they signal success, add a health-check verification by using the
+     * cfn-init helper script. For an example, see the verify_instance_health
+     * command in the Auto Scaling rolling updates sample template.
+     *
+     * @default true if you specified the minSuccessfulInstancesPercent property, false otherwise
+     */
+    waitOnResourceSignals?: boolean;
+
+    /**
+     * Specifies the Auto Scaling processes to suspend during a stack update.
+     *
+     * Suspending processes prevents Auto Scaling from interfering with a stack
+     * update.
+     *
+     * @default HealthCheck, ReplaceUnhealthy, AZRebalance, AlarmNotification, ScheduledActions.
+     */
+    suspendProcesses?: ScalingProcess[];
+}
+
+export enum ScalingProcess {
+    Launch = 'Launch',
+    Terminate = 'Terminate',
+    HealthCheck = 'HealthCheck',
+    ReplaceUnhealthy = 'ReplaceUnhealthy',
+    AZRebalance = 'AZRebalance',
+    AlarmNotification = 'AlarmNotification',
+    ScheduledActions = 'ScheduledActions',
+    AddToLoadBalancer = 'AddToLoadBalancer'
+}
+
+/**
+ * Render the rolling update configuration into the appropriate object
+ */
+function renderRollingUpdateConfig(config: RollingUpdateConfiguration = {}): cdk.AutoScalingRollingUpdate {
+    const waitOnResourceSignals = config.minSuccessfulInstancesPercent !== undefined ? true : false;
+    const pauseTimeSec = config.pauseTimeSec !== undefined ? config.pauseTimeSec : (waitOnResourceSignals ? 300 : 0);
+
+    return {
+        maxBatchSize: config.maxBatchSize,
+        minInstancesInService: config.minInstancesInService,
+        minSuccessfulInstancesPercent: validatePercentage(config.minSuccessfulInstancesPercent),
+        waitOnResourceSignals,
+        pauseTime: renderIsoDuration(pauseTimeSec),
+        suspendProcesses: config.suspendProcesses !== undefined ? config.suspendProcesses :
+            // Recommended list of processes to suspend from here:
+            // https://aws.amazon.com/premiumsupport/knowledge-center/auto-scaling-group-rolling-updates/
+            [ScalingProcess.HealthCheck, ScalingProcess.ReplaceUnhealthy, ScalingProcess.AZRebalance,
+                ScalingProcess.AlarmNotification, ScalingProcess.ScheduledActions],
+    };
+}
+
+/**
+ * Render a number of seconds to a PTnX string.
+ */
+function renderIsoDuration(seconds: number): string {
```

Comment:
Is `PT<seconds>S` not okay if the amount of seconds is large? I'm (very) mildly surprised you have to go normalize like that...

Reply:
Did not test, but I fully expect the validator on the other end to be strict about it.